src/Tests/Microsoft.NET.ToolPack.Tests/GivenThatWeWantToPackAToolProject.cs

```diff
@@ -0,0 +1,149 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Microsoft.NET.TestFramework;
+using Microsoft.NET.TestFramework.Assertions;
+using Microsoft.NET.TestFramework.Commands;
+using FluentAssertions;
+using Xunit;
+using Xunit.Abstractions;
+using NuGet.Packaging;
+
+namespace Microsoft.NET.ToolPack.Tests
+{
+    public class GivenThatWeWantToPackAToolProject : SdkTest
+    {
+        private readonly string _nugetPackage;
+
+        public GivenThatWeWantToPackAToolProject(ITestOutputHelper log) : base(log)
+        {
+            TestAsset helloWorldAsset = _testAssetsManager
+                .CopyTestAsset("PortableTool", "PackPortableTool")
+                .WithSource()
+                .Restore(Log);
+
+            var packCommand = new PackCommand(Log, helloWorldAsset.TestRoot);
+
+            packCommand.Execute();
+
+            _nugetPackage = packCommand.GetNuGetPackage();
+        }
+
+        [Fact]
+        public void It_packs_successfully()
+        {
+            using (var nupkgReader = new PackageArchiveReader(_nugetPackage))
+            {
+                nupkgReader
+                    .GetToolItems()
+                    .Should().NotBeEmpty();
+            }
+        }
+
+        [Fact]
+        public void It_finds_the_entry_point_dll_and_command_name_and_put_in_setting_file()
+        {
+            using (var nupkgReader = new PackageArchiveReader(_nugetPackage))
+            {
+                IEnumerable<NuGet.Frameworks.NuGetFramework> supportedFrameworks = nupkgReader.GetSupportedFrameworks();
+                supportedFrameworks.Should().NotBeEmpty();
+
+                var tmpfilePath = Path.Combine(Path.GetTempPath(), Path.GetRandomFileName());
+                string copiedFile = nupkgReader.ExtractFile($"tools/DotnetToolSettings.xml", tmpfilePath, null);
+                File.ReadAllText(copiedFile)
+                    .Should()
+                    .Contain("consoledemo.dll", "it should contain entry point dll that is same as the msbuild well known properties $(TargetFileName)")
+                    .And
+                    .Contain("consoledemo", "it should contain command name that is same as the msbuild well known properties $(TargetName)");
+            }
+        }
+
+        [Fact(Skip = "Pending https://github.com/NuGet/Home/issues/6354")]
+        public void It_adds_platform_package_to_dependency_and_remove_other_package_dependency()
```

Comment:
This is a must have. But we are pending NuGet's change. I am not sure I should check in before this is done. That is the reason I put WIP in the title

Reply:
find a way to solve that